### PR TITLE
CR-150: adding Summary job to Reach 

### PR DESCRIPTION
**Pull request by MosheMaorKaltura**

### CODE REVIEW

File: plugins/reach/lib/model/VendorCatalogItem.php [hunk 1]
```diff
@@ -253,7 +253,10 @@ public static function translateServiceFeatureEnum($catalogItemId)
 				$serviceFeatureName = 'quiz';
 				break;
 
-			case VendorServiceFeature::OCR:
+			case VendorServiceFeature::SUMMARY:
+				$serviceFeatureName = 'summary';
+
+      case VendorServiceFeature::OCR:
```
Comment (drorsou) [inline plugins/reach/lib/model/VendorCatalogItem.php:259]: Indentation

File: plugins/reach/lib/model/data/kSummaryVendorTaskData.php [hunk 2]
```diff
@@ -0,0 +1,63 @@
+<?php
+
+/**
+ * @package plugins.reach
+ * @subpackage model
+ */
+class kSummaryVendorTaskData extends kVendorTaskData
+{
+	public String $typeOfSummary;
```
Comment (drorsou) [inline plugins/reach/lib/model/data/kSummaryVendorTaskData.php:9]: Core class fields should be _protected_

File: plugins/reach/lib/api/data/entryVendorTaks/KalturaSummaryVendorTaskData.php [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+		return parent::toObject($dbObject, $propsToSkip);
+	}
+
+	public function validateForInsert($propertiesToSkip = array())
+	{
+		$this->validatePropertyNotNull("typeOfSummary");
+		$this->validatePropertyNotNull("writingStyle");
+	}
+
+	public function getSummaryOutputJson(): ?string
```
Comment (drorsou) [inline plugins/reach/lib/api/data/entryVendorTaks/KalturaSummaryVendorTaskData.php:66]: No need for getters/setters, the class fields are public

File: plugins/reach/lib/api/data/vendorCatalogItem/KalturaVendorSummaryCatalogItem.php [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+class KalturaVendorSummaryCatalogItem extends KalturaVendorCatalogItem
+{
+	protected function getServiceFeature(): int
+	{
+		return VendorServiceFeature::SUMMARY;
+	}
+
+	public function toInsertableObject($object_to_fill = null, $props_to_skip = array())
+	{
+		if (is_null($object_to_fill)) {
```
Comment (drorsou) [inline plugins/reach/lib/api/data/vendorCatalogItem/KalturaVendorSummaryCatalogItem.php:15]: move { to newline

File: plugins/reach/lib/api/data/vendorCatalogItem/KalturaVendorSummaryCatalogItem.php [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+			$object_to_fill = new VendorSummaryCatalogItem();
+		}
+
+		return parent::toInsertableObject($object_to_fill, $props_to_skip);
+	}
+
+	public function toObject($object_to_fill = null, $props_to_skip = array())
+	{
+		if (is_null($object_to_fill)) {
```
Comment (drorsou) [inline plugins/reach/lib/api/data/vendorCatalogItem/KalturaVendorSummaryCatalogItem.php:24]: move { to newline

File: plugins/reach/lib/api/data/vendorCatalogItem/filters/KalturaVendorSummaryCatalogItemFilter.php [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+/**
+ * @package plugins.reach
+ * @subpackage api.filters
+ */
+class KalturaVendorSummaryCatalogItemFilter extends KalturaVendorCatalogItemFilter
+{
+	public function getTypeListResponse(KalturaFilterPager $pager, KalturaDetachedResponseProfile $responseProfile = null, $type = null)
+	{
+		if (!$type) {
```
Comment (drorsou) [inline plugins/reach/lib/api/data/vendorCatalogItem/filters/KalturaVendorSummaryCatalogItemFilter.php:11]: move { to newline

File: plugins/reach/admin/CatalogItemListAction.php [hunk 7]
```diff
@@ -81,6 +81,8 @@ protected function getCatalogItemFilterByServiceFeature($serviceFeature)
 			return new Kaltura_Client_Reach_Type_VendorClipsCatalogItemFilter();
 		elseif ($serviceFeature == Kaltura_Client_Reach_Enum_VendorServiceFeature::QUIZ)
 			return new Kaltura_Client_Reach_Type_VendorQuizCatalogItemFilter();
+	   elseif ($serviceFeature == Kaltura_Client_Reach_Enum_VendorServiceFeature::SUMMARY)
```
[the remark this answers is not in the document]
Comment (MosheMaorKaltura) [reply plugins/reach/admin/CatalogItemListAction.php]: Done

File: plugins/reach/lib/api/data/vendorCatalogItem/KalturaVendorCatalogItem.php [hunk 8]
```diff
@@ -294,7 +294,11 @@ public static function getInstance($sourceObject, KalturaDetachedResponseProfile
         
 			case VendorServiceFeature::QUIZ:
 				$object = new KalturaVendorQuizCatalogItem();
-        break;
+                break;
```
[the remark this answers is not in the document]
Comment (MosheMaorKaltura) [reply plugins/reach/lib/api/data/vendorCatalogItem/KalturaVendorCatalogItem.php]: Done

File: plugins/reach/lib/model/VendorCatalogItem.php (hunk 1 above)
Comment (MosheMaorKaltura) [reply plugins/reach/lib/model/VendorCatalogItem.php]: Done

File: plugins/reach/lib/model/data/kSummaryVendorTaskData.php (hunk 2 above)
Comment (MosheMaorKaltura) [reply plugins/reach/lib/model/data/kSummaryVendorTaskData.php]: done

**Review by drorsou — APPROVED** [review]

**Pull request merged**